### ISSUE TITLE
Error on macOS when using Experimental Scripting Runtime Version (.Net 4.6 Equivalent) and NAT punchthrough enabled

### ISSUE THREAD

**Issue by DMeville** [BoltEngine/Bolt-Tracker#61]
###How to reproduce issue
1. On macOS, download latest version of bolt from asset store. Import Photon and Photon Sample packages.
2. Change Scripting Runtime Version to Experimental (.NET 4.6 Equivalent) in the Player Settings.
3. Run PhotonTutorial, try to start server or client.

###Expected Behavior
Should startup properly.

###Actual behavior
Startup fails with the following error:
ArgumentException: Value does not fall within the expected range.
System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Int32 optionValue) (at <2d96311935af40c7b35c6ff5c955d7e7>:0)
DotNetSocket..ctor (DotNetPlatform platform, System.Boolean ipv6) (at <6bcdd95062d94a61a5c285641b670f40>:0)
udpkit.plataform.photon.puncher.PunchSocket..ctor (DotNetPlatform platform, System.Boolean ipv6) (at <99660c26366c41fc8524ccf52ede00a5>:0)
udpkit.plataform.photon.puncher.PunchPlatform.CreateSocket (System.Boolean ipv6) (at <99660c26366c41fc8524ccf52ede00a5>:0)
PhotonPlatform.CreateSocket (System.Boolean ipv6) (at <99660c26366c41fc8524ccf52ede00a5>:0)
UdpKit.UdpSocket..ctor (System.Guid gameId, UdpKit.UdpPlatform p, UdpKit.UdpConfig cfg) (at <d6acf1eabd3541adb7dd5adb2e3d9d58>:0)
BoltCore.BeginStart (Bolt.ControlCommandStart cmd) (at <ca259606159d40b081ef7af8f5070eeb>:0)
Bolt.ControlCommandStart.Run () (at <ca259606159d40b081ef7af8f5070eeb>:0)
Bolt.ControlBehaviour.Update () (at <ca259606159d40b081ef7af8f5070eeb>:0)
UnityEngine.Debug:LogException(Exception)
Bolt.ControlBehaviour:Update()

### Additional notes
Everything works properly on Windows, this error only seems to happen on macOS (and maybe Linux, haven't tested yet)

Disabling "NAT Punchthrough Enabled" from the bolt settings windows and recompiling assembly and testing allows things to startup properly.  

Using the "Experimental (.NET 4.6 Equivalent) Scripting Runtime Version is required when using [Facepunch.Steamworks](https://github.com/Facepunch/Facepunch.Steamworks).

###Configuration
Unity Version: 2017.3.0p1
Bolt Version 1.1.0.5
Operating System: macOS 10.12.6 (Sierra)
Network Configuration: N/A

**Comment by aidanwolf** [comment]
```
ArgumentException: Value does not fall within the expected range.
System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Int32 optionValue) (at <4ca7fa06156443beb88ec302e6e9783c>:0)
DotNetSocket..ctor (DotNetPlatform platform, System.Boolean ipv6) (at <6bcdd95062d94a61a5c285641b670f40>:0)
DotNetPlatform.CreateSocket (System.Boolean ipv6) (at <6bcdd95062d94a61a5c285641b670f40>:0)
UdpKit.UdpSocket..ctor (System.Guid gameId, UdpKit.UdpPlatform p, UdpKit.UdpConfig cfg) (at <d6acf1eabd3541adb7dd5adb2e3d9d58>:0)
BoltCore.BeginStart (Bolt.ControlCommandStart cmd) (at <ca259606159d40b081ef7af8f5070eeb>:0)
Bolt.ControlCommandStart.Run () (at <ca259606159d40b081ef7af8f5070eeb>:0)
Bolt.ControlBehaviour.Update () (at <ca259606159d40b081ef7af8f5070eeb>:0)
UnityEngine.Debug:LogException(Exception)
Bolt.ControlBehaviour:Update()
```

MacOS 10.13.5 (17F45c)
Unity 2018.2.0b1
NET 4.x Equivalent

Rolling back to NET 3.5 fixes error, but not a solution

**Comment by aidanwolf** [comment]
2018.1 is no longer in beta as of today. When can we expect a patch? Thank you

**Comment by DMeville** [comment]
@ramonmelo The updated version of udpkit.platform.dotnet.dll you sent to me appears to have fixed the issue in 2017.3.0p1 when using the "Experimental (.NET 4.6 Equivalent) Scripting Runtime Version. 

I have not yet tested this on 2018.1.

**Comment by ramonmelo** [comment]
Fixed on the next release
